tensorflow/python/keras/integration_test/gradient_checkpoint_test.py

```diff
@@ -0,0 +1,163 @@
+# limitations under the License.
+# ==============================================================================
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.keras import layers, optimizers
+
+def _get_big_cnn_model(img_dim, n_channels, num_partitions,
+                       blocks_per_partition):
+  """Creates a test model whose activations are significantly larger than model size."""
+  model = tf.keras.Sequential()
+  model.add(layers.Input(shape=(img_dim, img_dim, n_channels)))
+  for _ in range(num_partitions):
+    for _ in range(blocks_per_partition):
+      model.add(layers.Conv2D(10, 5, padding='same', activation=tf.nn.relu))
+      model.add(layers.MaxPooling2D((1, 1), padding='same'))
+      model.add(layers.Conv2D(40, 5, padding='same', activation=tf.nn.relu))
+      model.add(layers.MaxPooling2D((1, 1), padding='same'))
+      model.add(layers.Conv2D(20, 5, padding='same', activation=tf.nn.relu))
+      model.add(layers.MaxPooling2D((1, 1), padding='same'))
+  model.add(layers.Flatten())
+  model.add(layers.Dense(32, activation=tf.nn.relu))
+  model.add(layers.Dense(10))
+  return model
+
+
+def _get_split_cnn_model(img_dim, n_channels, num_partitions,
+                         blocks_per_partition):
+  """Creates a test model that is split into `num_partitions` smaller models"""
+  models = [tf.keras.Sequential() for _ in range(num_partitions)]
+  models[0].add(layers.Input(shape=(img_dim, img_dim, n_channels)))
+  for i in range(num_partitions):
+    model = models[i]
+    if i > 0:
+      last_shape = models[i - 1].layers[-1].output_shape
+      model.add(layers.Input(shape=last_shape[1:]))
+    for _ in range(blocks_per_partition):
+      model.add(layers.Conv2D(10, 5, padding='same', activation=tf.nn.relu))
+      model.add(layers.MaxPooling2D((1, 1), padding='same'))
+      model.add(layers.Conv2D(40, 5, padding='same', activation=tf.nn.relu))
+      model.add(layers.MaxPooling2D((1, 1), padding='same'))
+      model.add(layers.Conv2D(20, 5, padding='same', activation=tf.nn.relu))
+      model.add(layers.MaxPooling2D((1, 1), padding='same'))
+  models[-1].add(layers.Flatten())
+  models[-1].add(layers.Dense(32, activation=tf.nn.relu))
+  models[-1].add(layers.Dense(10))
+  return models
+
+
+def _compute_loss(logits, labels):
+  return tf.reduce_mean(
+      tf.nn.sparse_softmax_cross_entropy_with_logits(logits=logits,
+                                                     labels=labels))
+
+def _limit_gpu_memory():
+  """Helper function to limit GPU memory for testing  """
+  gpus = tf.config.experimental.list_physical_devices('GPU')
+  if gpus:
+    try:
+      tf.config.experimental.set_virtual_device_configuration(
+          gpus[0], [
+              tf.config.experimental.VirtualDeviceConfiguration(
+                  memory_limit=1024)
+          ])
+    except RuntimeError as e:
+      print(e)
```

Comment:
What kind of RuntimeError is being ignored here? And why is this non-fatal?

The internal CI doesn't like `print`; that's easy to ignore, but I'm wondering if there's a more explicit thing we can do instead.

Reply:
You are right. I think the test should either fail or be skipped if a virtual GPU cannot be created. Should I add a return False in which case the test will be skipped?

Reply:
If we don't think it'll happen during CI builds, I'm happy with this being a test failure. I don't see why it would: either it'll have a decent-sized GPU or it won't have anything.

Reply:
Ok. I will just remove the try/except block then and update the PR?

Reply:
Sounds good to me.